File: src/main/java/dev/vixid/vsm/config/SpotifyConfig.java
package dev.vixid.vsm.config;

import com.google.gson.annotations.Expose;
import dev.vixid.vsm.config.core.Position;
import dev.vixid.vsm.config.core.annotations.ConfigEditorKeybind;
import dev.vixid.vsm.overlays.OverlayPositions;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorButton;
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;

public class SpotifyConfig {

    @Expose
    @ConfigOption(name = "Enable", desc = "Enable Spotify Overlay")
    @ConfigEditorBoolean
    public boolean enabled = true;

    @Expose
    public Position overlayPosition = new Position(10, 10);

    @Expose
    @ConfigOption(name = "Skip Forward", desc = "Key bind to skip forward a song")
    @ConfigEditorKeybind(defaultKey = "key.keyboard.unknown")
    public String skipForwardKey = "key.keyboard.unknown";

    @Expose
    @ConfigOption(name = "Skip Backward", desc = "Key bind to skip backward a song")
    @ConfigEditorKeybind(defaultKey = "key.keyboard.unknown")
    public String skipBackwardKey = "key.keyboard.unknown";

    @Expose
    @ConfigOption(name = "Play / Pause", desc = "Key bind to play / pause a song")
    @ConfigEditorKeybind(defaultKey = "key.keyboard.unknown")
    public String playPauseKey = "key.keyboard.unknown";

    @ConfigOption(name = "Edit Song Location", desc = "Change the position of the song overlay")
    @ConfigEditorButton(buttonText = "Edit")
    public Runnable editPosition = OverlayPositions.INSTANCE::openPositionEditor;
}

File: src/main/kotlin/dev/vixid/vsm/config/SpotifyConfig.kt
package dev.vixid.vsm.config

import com.google.gson.annotations.Expose
import dev.vixid.vsm.config.core.Position
import dev.vixid.vsm.overlays.OverlayPositions
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorButton
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorKeybind
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption
import org.lwjgl.glfw.GLFW

class SpotifyConfig {

    @Expose
    @ConfigOption(name = "Enable", desc = "Enable Spotify Overlay")
    @ConfigEditorBoolean
    var enabled = true

    @Expose
    val position: Position = Position(10, 10)

    @Expose
    @ConfigOption(name = "Skip Forward", desc = "Keybind to skip forward a song")
    @ConfigEditorKeybind(defaultKey = GLFW.GLFW_KEY_UNKNOWN)
    var skipForwardKey = GLFW.GLFW_KEY_UNKNOWN

    @Expose
    @ConfigOption(name = "Skip Backward", desc = "Keybind to skip backward a song")
    @ConfigEditorKeybind(defaultKey = GLFW.GLFW_KEY_UNKNOWN)
    var skipBackwardKey = GLFW.GLFW_KEY_UNKNOWN

    @Expose
    @ConfigOption(name = "Play / Pause", desc = "Keybind to play / pause a song")
    @ConfigEditorKeybind(defaultKey = GLFW.GLFW_KEY_UNKNOWN)
    var playPauseKey = GLFW.GLFW_KEY_UNKNOWN

    @ConfigOption(name = "Edit Song Location", desc = "Change the location of the song name on screen")
    @ConfigEditorButton(buttonText = "Edit")
    val editPosition: Runnable = Runnable(OverlayPositions::openPositionEditor)
}